### DISCUSSION TITLE
OfflineAudioContext tail time when rendering a reverb effect

### DISCUSSION THREAD

**Discussion by arturogalan** [WebAudio/web-audio-api#2557] [Q&A] [answered]
Hello everybody, I'm trying to implement something with OfflineAudioContext but I'm stuck on this problem:

1-I have a reverb effect AudioWorkletProcessor with a decay value that repeats the input sound several times like an echo, what it's called tail time.
  
2-If I use a new AudioContext, connect the reverb node, and reproduce a track then I can hear the original audio + the tail time that the effect adds to the original track length with the echo effect. The audio reproduces until the effect stops rendering the echo
  

`const track = audioContext.createMediaElementSource(audioElement);
  track.connect(gainNode).connect(reverbNode).connect(audioContext.destination);
 track.mediaElement.play()
`
  
3-BUT when I use the OfflineAudioContext to try to get the track + reverb effect applied, I can listen to the rendered result audio but the track stops at the original length of the track, so I cannot hear all the effect tail time. It stops rendering before the effect ends

```
const offlineCtx = new OfflineAudioContext(
    decodedBuffer.numberOfChannels,
    decodedBuffer.length,
    decodedBuffer.sampleRate
  );  
  const reverbNodeOff = new AudioWorkletNode(offlineCtx, "reverb-processor");  

const source = new AudioBufferSourceNode(offlineCtx, {
    buffer: decodedBuffer,
  });  

  source.connect(reverbNodeOff).connect(offlineCtx.destination);  

  source.start() 

  const renderedBuffer = await offlineCtx.startRendering()  

  const song = new AudioBufferSourceNode(audioCtx, {
    buffer: renderedBuffer,
  });  
  song.connect(audioCtx.destination);  
  song.start();  
```

Is there a way to tell the OfflineAudioContext that should render until the effect stops?
I don't know if the problem is in the OfflineAudioContext.render, or I should add some logic in the definition of the AudioWorkletProcessor effect node
Please tell me if you need more context and thank you!

**Comment by chrisguttandin** [comment, marked as answer]
You would probably need to increase the `length` of your `OfflineAudioContext` by the number of samples of the tail time.

```js
new OfflineAudioContext({
    length: decodedBuffer.length + TAIL_TIME_IN_SAMPLES,
    numberOfChannels: decodedBuffer.numberOfChannels,
    sampleRate: decodedBuffer.sampleRate
});
```

**Comment by arturogalan** [comment]
I tested that and the song duration is increased, but with silence
